### DISCUSSION TITLE
How to have a fixed dimensional latent space after entropy bottleneck.compress()?

### DISCUSSION THREAD

**Discussion by PatatalouiS** [InterDigitalInc/CompressAI#150] [Q&A]
Hello everyone,

I'm looking for trained Auto-encoder architectures, to reduce the size of video frames as much as possible, before feeding another architecture, for a rather specific video indexing task by content.

CompressAi fits my needs perfectly. I want to get the output string, to send it to my other network. After converting the string into an integer representation, I realize that the length of this space is not fixed and depends on the input image. This must be a particularity of entropy coding.

Do you have an idea, with a minimal loss of information, for each latent space to be of fixed size? 

Maybe with an operation on the encoding provided by compress()?
Maybe by replacing the entropy_bottleneck by another encoding method?

Thanks in advance, and congratulations for this splendid library.

**Comment by YodaEmbedding** [comment]
As you mentioned, the latent space `y = g_a(x)` has a height and width that are 16x (or some other factor) smaller than the input. This shape must be transmitted to the decoder.
